software/control/core.py

```diff
@@ -801,9 +847,7 @@ def scan_preview_move_from_click(self, click_x, click_y, image_width, image_heig
         pixel_sign_y = -1 if INVERTED_OBJECTIVE else 1
 
         # move to selected fov
-        self.move_x_to(self.scan_begin_position_x+dx_mm*fov_col*pixel_sign_x)
-        self.microcontroller.wait_till_operation_is_completed()
-        self.move_y_to(self.scan_begin_position_y+dy_mm*fov_row*pixel_sign_y)
+        self.move_to(self.scan_begin_position_x+dx_mm*fov_col*pixel_sign_x, self.scan_begin_position_y+dy_mm*fov_row*pixel_sign_y)
         self.microcontroller.wait_till_operation_is_completed()
```

Comment:
this line should be deleted after you switch to move_to(), which is blocking. At a later point we can better organize these move functions to make their implementation more consistent.